### ISSUE TITLE
Building manaplus

### ISSUE THREAD

**Issue by symbios24** [ManaPlus/ManaPlus#30]
Hello im trying to build manaplus for the Raspberry Pi 3 by executing a script that i found here

https://github.com/zerojay/RetroPie-Extra/blob/master/scriptmodules/ports/manaplus.sh

it take`s sometime to clone manaplus but once its finish it does not install it it exits

the script was working a year ago but now its not building manaplus

can you help me by looking the script and tell me if anything is changed by then

Thanks

**Comment by 4144** [comment]
i think nothing changed in manaplus itself, but can be changed in distro what you using.
can you show error messages?

**Comment by symbios24** [comment]
i searched for error messages but could not find any,im using raspbian stretch,i can install manaplus from apt-get install manaplus but its an old version eg 1.7

**Comment by 4144** [comment]
try run ``./manaplus.sh 2>err.txt``
after script complete, show ``err.txt``

**Comment by symbios24** [comment]
the command ./manaplus.sh 2>err.txt
does not do anything

**Comment by 4144** [comment]
manaplus.sh is your script what you show in first message

**Comment by symbios24** [comment]
yes the script in this location

https://github.com/zerojay/RetroPie-Extra/blob/master/scriptmodules/ports/manaplus.sh

**Comment by 4144** [comment]
probably some thing changed in retropie scripts usage. try ask on them how to use it or search docs.

but if you have all manaplus dependencies installed, you can just clone and build manaplus
see inside manaplus file ``build.sh``

**Comment by symbios24** [comment]
Can you please put in your github page some basic building instructions under the dependencies? coz i dont really understand what to type from build.sh

Thank you

**Comment by 4144** [comment]
dependencies different for each distro. lib names already present in readme

from retroie script need install this packages ``libsdl1.2-dev libsdl-mixer1.2-dev libsdl-image1.2-dev libsdl-net1.2-dev libcurl4-gnutls-dev libpng12-dev libxml2-dev automake libguichan-dev patch autoconf libtool libsdl-ttf2.0-dev libsdl-gfx1.2-dev gettext libgl1-mesa-dev libenet-dev libphysfs-dev libxml2-dev zlib1g-dev xorg``

Some packages in this list useless, but installing them all will not create issues.

**Comment by symbios24** [comment]
i mean to put after the dependencies how to build manaplus

**Comment by 4144** [comment]
simply run ``build.sh`` it do all the rest.
will build and install manaplus into current subdir with name ``run``

**Comment by symbios24** [comment]
do i need to clone it first?  eg git clone https://gitlab.com/manaplus/manaplus.git

**Comment by 4144** [comment]
yes

**Comment by symbios24** [comment]
thanks i will try it tomorrow and let you know

**Comment by symbios24** [comment]
the autopoint package from the depends was missing that why the install was failing

**Comment by symbios24** [comment]
it installed successfully,i tried to run it with manaplus and all i see is a blue arrow showing up after i see some graphics distortion and thats it i tried also the manaplus -safemode command i got the same result
i also tried it with the xinit command i got the same any suggestions?

**Comment by symbios24** [comment]
Another thing i encounter is that once in game i set the resolution for example to 1024x768 i press apply and exit the game
the next time i run the game the resolution is set to default eg 640x480 
do you have anything in mind about that?

thanks

**Comment by 4144** [comment]
blue arrows is hardware test, you should wait a bit
often drives or video cards slow or not working correct, because this manaplus do some graphics tests and using best mode what it detecting

**Comment by symbios24** [comment]
i found in the changelog that the default resolution for safe mode is 640x480 can this be changed somehow?

**Comment by 4144** [comment]
no. safe mode mean drop almost all settings and run somehow client anyway if all broken.
better do normal hardware test, and after it should works fine

**Comment by symbios24** [comment]
how i do the hardware test?

**Comment by 4144** [comment]
run manaplus normally. it will show moving blue arrows. window title will be "VIDEO MODE TEST"
wait until it complete

**Comment by symbios24** [comment]
i did that the screen stayed blank and did not come up
can you add in your next commit the minimum safe mode resolution to be 1024x768 or 800x600 ??
thanks

**Comment by 4144** [comment]
blank you mean it show blank window?
then need wait, probably because cpu/gpu very slow it working bit longer than usual
for see progress you can try run manaplus from terminal

**Comment by symbios24** [comment]
i run it from terminal by default otherwise i cannot put the the safemode command and if i run it normal i only see distorted graphics

**Comment by symbios24** [comment]
so i will run it normal and wait until is finished or i run it with the --tests or renderer command ??

**Comment by 4144** [comment]
probably better run normally

**Comment by symbios24** [comment]
i left it half a hour nothing happened anyway thanks for the help if you can in the future to add a commit with a higher safemode resolution will be great

**Comment by symbios24** [comment]
at least please release in a commit 800x600 as the default safe mode res? 
nobody can play with that small res and everyone has a modern monitor nowadays

**Comment by 4144** [comment]
Btw in your system performance tests may stuck because you have some bad opengl implementation.
Try build manaplus without opengl and run performance tests again

Alternative way is crack configuration file. But in this way OpenGL modes may not works.
Close all client instances,
in file ``config.xml`` search for option ``videodetected`` if if exists, change value to ``1``. If not exists add line ``<option name="videodetected" value="1"/>``.

**Comment by symbios24** [comment]
thanks but the opengl is not a problem it works even in safe mode,the problem is that the game only runs in safe mode,can you add a command to run it in safe mode with another resolution for example
--safemode -res 1024x768

**Comment by 4144** [comment]
no. safe mode is safe mode. it disable all possible features and try run client,
if you dont want mess with compilation, follow alternative way from my previous message

**Comment by 4144** [comment]
also you cant use opengl in safe mode. Safe mode disable ALL possible feature/modes. it left no optimizations, no opengl and many other no things.

You should fix your hardware or os or disable auto detection manually and use software mode.

**Comment by symbios24** [comment]
How can i build the game without opengl i cannot seem to find any instructions in your repository

**Comment by pazkero** [comment]
./configure --without-opengl

As o11c would say... "I will not dump the whole ./configure --help here"
